lib/pool.js

```diff
@@ -17,49 +21,104 @@ class Pool {
       throw new InvalidArgumentError('invalid connections')
     }
 
+    this[kQueue] = []
+    this[kPendingIdx] = 0
     this[kClients] = Array.from({
       length: connections || 10
     }, () => new Client(url, options))
+
+    for (const client of this[kClients]) {
+      client.on('drain', () => {
```

Comment:
can you please create this closure in a separate function? closures and loops are.. weird.

Reply:
it needs to capture the client instance though?

Reply:
you don't need it, it is `this` in the function (if you remove the arrow).

Reply:
Yes, something like `client.on('drain', onDrain(this))`

Reply:
How about now?